### PR TITLE
use real alert data to determine flake and failure timeframes for alerts

### PR DESCRIPTION
**Pull request by deads2k**
This will prevent us from getting worse without noticing it, but it will not force an improvement.

### CODE REVIEW

File: pkg/synthetictests/allowedalerts/types.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  alertName = "etcdHighNumberOfLeaderChanges" or 
+  alertName = "KubeAPIErrorBudgetBurn" or 
+  alertName = "KubeClientErrors" or 
+  alertName = "KubePersistentVolumeErrors" or 
+  alertName = "MCDDrainError" or 
+  alertName = "PrometheusOperatorWatchErrors"
+order by 
+ AlertName, Release, FromRelease, Topology, Platform, Network
+`
+)
```
Comment (dgoodwin) [inline pkg/synthetictests/allowedalerts/types.go:32]: This appears unused. How does one update the json file? We'd need that documented somewhere, and I do not know where that somewhere should be. It's a thing that someone should do periodically right? 

I'm actually a little worried about this going in hard-coded at all here, will we struggle with this getting out of date and not matching what's going on in aggregation? What are your thoughts around maintenance of this.

Pie in the sky, someday, what if this data was exposed in sippy via an unathenticated json API, and the test could hit that endpoint to fetch latest data.

**Review by dgoodwin — COMMENTED** [review]
Looks like a big improvement just need to hear your thoughts on maintenance and documentation.

**Comment by deads2k** [comment]
I've updated comments to explain how to build this.  I am in favor of creating a command to generate these json files that lives in openshift/origin as a jira card.  

I'm going to get this merged to clean up where we are, but I will address additional comments as followups if you leave them.

**Pull request merged**